V2DriverDemo.md

```diff
@@ -0,0 +1,352 @@
+
+#Set the parameters
+LOCATION=northeurope # Location 
+AKS_NAME=az-zrs
+RG=$AKS_NAME-$LOCATION
+AKS_CLUSTER_NAME=$AKS_NAME-cluster # name of the cluster
+K8S_VERSION=$(az aks get-versions  -l $LOCATION --query 'orchestrators[-1].orchestratorVersion' -o tsv)
+
+
+##Create RG
+az group create --name $RG --location $LOCATION
+
+
+## create the cluster 
+az aks create \
+-g $RG \
+-n $AKS_CLUSTER_NAME \
+-l $LOCATION \
+--kubernetes-version $K8S_VERSION \
+--zones 1 2 3 \
+--generate-ssh-keys 
+
+
+## get the credentials 
+
+az aks get-credentials -n $AKS_CLUSTER_NAME -g $RG
+
+## test
+
+kubectl get nodes  
+
+NAME                                STATUS   ROLES   AGE   VERSION
+aks-nodepool1-20996793-vmss000000   Ready    agent   77s   v1.21.2
+aks-nodepool1-20996793-vmss000001   Ready    agent   72s   v1.21.2
+aks-nodepool1-20996793-vmss000002   Ready    agent   79s   v1.21.2
+```
+
+2. Install the Azure Disk CSI Driver v2
+
+As of K8s 1.21 CSI became the default in storage drivers, same in AKS, you can see the default storage class now pointing to v1 azure disk CSI driver. To demonstrate the Azure Disk CSI Driver V2, first we will install it such that it runs side by side with the V1 driver. Tips for troubleshooting helm installation [here](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/668a54a797fd90f015ce2b89ff2fbac2d0a4600b/charts/README.md).
+
+```shell
+helm install azuredisk-csi-driver-v2 charts/v2.0.0-alpha.1/azuredisk-csi-driver --namespace kube-system --wait --timeout=15m -v=5 
+```
+
+
+Verify that the new storage class was created
+```shell
+
+kubectl get storageclasses.storage.k8s.io 
+
+NAME                    PROVISIONER                RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
+azurefile               kubernetes.io/azure-file   Delete          Immediate              true                   2m30s
+azurefile-csi           file.csi.azure.com         Delete          Immediate              true                   2m30s
+azurefile-csi-premium   file.csi.azure.com         Delete          Immediate              true                   2m30s
+azurefile-premium       kubernetes.io/azure-file   Delete          Immediate              true                   2m30s
+csi-v2                  disk2.csi.azure.com        Delete          Immediate              true                   1m30s
+default (default)       disk.csi.azure.com         Delete          WaitForFirstConsumer   true                   2m30s
+managed                 kubernetes.io/azure-disk   Delete          WaitForFirstConsumer   true                   2m30s
+managed-csi-premium     disk.csi.azure.com         Delete          WaitForFirstConsumer   true                   2m30s
+managed-premium         kubernetes.io/azure-disk   Delete          WaitForFirstConsumer   true                   2m30s
+```
+
+Keep in mind that if you are creating your own storage class to utilize the V2 driver, the parameter for `maxShares` must be > 1 for there to be replicas
```

Comment:
np: "...must be > 1 to benefit from faster pod failover through replica mounts."